src/Bicep.LangServer/Telemetry/TelemetryProvider.cs

```diff
@@ -0,0 +1,25 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using OmniSharp.Extensions.LanguageServer.Protocol.Server;
+using OmniSharp.Extensions.LanguageServer.Protocol.Window;
+
+namespace Bicep.LanguageServer.Telemetry
+{
+    public class TelemetryProvider : ITelemetryProvider
+    {
+        public ILanguageServer? LanguageServer { get; set; }
+
+        public void PostEvent(TelemetryEvent telemetryEvent)
+        {
+            if (telemetryEvent is null ||
+                !string.IsNullOrWhiteSpace(telemetryEvent.EventName) ||
+                telemetryEvent.Properties?.Count == 0)
+            {
+                return;
+            }
+
+            LanguageServer?.Window.SendTelemetryEvent(telemetryEvent);
```

Comment:
FYI, another benefit to using constructor injection is that you don't have to handle the possibility that `LanguageServer` could be `null`.

Reply:
Updated here:
https://github.com/Azure/bicep/commit/ffbef4cf981f55f6966f50de6ed4dec09bd90905